app/views/usergroups/_form.html.erb

```diff
@@ -46,7 +46,8 @@
         <button class='btn btn-info' type='button'
           onclick="$('#current_external_usergroups').toggle();"><%= _("Show linked external user groups") %></button>
         <hr>
-        <div style='display:none' id='current_external_usergroups'>
+        <% invalid_external_groups = @usergroup.external_usergroups.any? { |e| e.errors.present? } %>
+        <div style='<%= invalid_external_groups ? '' : 'display:none' -%>' id='current_external_usergroups'>
```

Comment:
-%> is moot since Rails 3, %> is preferred -> http://projects.theforeman.org/projects/foreman/wiki/Reviewing_patches

Other than that :+1: , works nicely